trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/expressiontranslation/IntegerTranslation.java

```diff
@@ -802,4 +783,88 @@ protected Pair<Expression, Expression> constructOverflowCheckForLeftShift(final
 			final ExpressionResult exprResult) {
 		return constructOverflowCheck(loc, resultType, exprResult.getLrValue().getValue());
 	}
+
+	@Override
+	public Expression roundToIntegral(final ILocation loc, final Expression argument, final CPrimitive type,
+			final Expression roundingMode) {
+		throw new UnsupportedOperationException(NOT_IMPLEMENTED);

```

Comment:
Would it (in the future, not this PR) make sense to implement some of these functions with overapproximations as well? Or do we prefer to fail early if these are used, e.g. because we assume only bitvector mode will have a chance to verify such programs? (CC @danieldietsch ?)

Reply:
Oh I think in particular for float we could have various different implementations, also with overapproximations. I don't think we want to fail early (or: at least not unconditionally), but rather always have the option of, e.g., havoc + assume inside type range + overapprox annotation.